pkg/daemon/ceph/osd/kms/azure.go

```diff
@@ -0,0 +1,77 @@
+package kms
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/libopenstorage/secrets"
+	"github.com/libopenstorage/secrets/azure"
+	"github.com/pkg/errors"
+	"github.com/rook/rook/pkg/clusterd"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// IsAzure determines whether the configured KMS is Azure Key Vault
+func (c *Config) IsAzure() bool {
+	return c.Provider == secrets.TypeAzure
+}
+
+// InitAzure initializes azure key vault client
+func InitAzure(ctx context.Context, context *clusterd.Context, namespace string, config map[string]string) (secrets.Secrets, error) {
+
+	azureKVConfig, removecertFiles, err := azureKVCert(ctx, context, namespace, config)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to setup azure client cert authentication")
+	}
+	defer removecertFiles()
+
+	c := make(map[string]interface{})
+	for key, value := range azureKVConfig {
+		c[key] = string(value)
+	}
+
+	secrets, err := azure.New(c)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to initialize azure client")
+	}
+
+	return secrets, nil
+}
+
+// azureKVCert retrivies azure client cert from the secret and stores that in a file
+func azureKVCert(ctx context.Context, context *clusterd.Context, namespace string, config map[string]string) (newConfig map[string]string, removeCertFiles removeCertFilesFunction, retErr error) {
+	var filesToRemove []*os.File
+	defer func() {
+		removeCertFiles = getRemoveCertFiles(filesToRemove)
+		if retErr != nil {
+			removeCertFiles()
+			removeCertFiles = nil
+		}
+	}()
+
+	clientCertSecretName := config["AZURE_CERT_SECRET_NAME"]
```

Comment:
Should this value be hardcoded if we are using the value that comes from libopenstorage/secrets?

Reply:
`AZURE_CERT_SECRET_NAME` is the name of the k8s secret that contains the Azure client certificate. Its specific to rook and is not coming from libopenstorage/secrets.